### DISCUSSION TITLE
Updating many to many relationships not possible?

### DISCUSSION THREAD

**Discussion by remcohh** [Budibase/budibase#5870] [General]
I have a self hosted installation connected to a postgres database.
When I define a many 2 many relation between 2 tables budi generates a link table which works fine when creating a record with relations. However It does not seem to be possbile to update those relations.
sometimes I get an error: "error: duplicate key value violates unique constraint "jt_rehearsals_ensembles_ensembles_rehearsals_pkey"
but sometimes don't get the error but it just wont update the relations. The other fields are updated just fine. Is this a know limitation?

**Comment by remcohh** [comment]
I investigated it a bit further with a small app (cars <-->artists) and it seems like a bug.
I first connect 2 cars with an artist, and after that i try to remove one.
In the screenshot the network request. As you can see one car (id) is posted but 2 car names.

![image](https://user-images.githubusercontent.com/1011465/168272494-42e680a1-2a48-4293-980e-8e59a3ac5a6c.png)

**Reply by melohagan** [reply]
HI @remcohh I wasn't able to replicate the issue. Are you updating from the Data tab in the builder, or through buttons on the running apps' UI?

Could you provide steps to reproduce the problem please?

**Reply by remcohh** [reply]
Hi @melohagan I created 2 new tables with a postgres datasource and in 1 table I added a relation column linking to the other table. This works fine when creating records but not updating. I see the problem in the forms I generated but also from the data tab in the builder. If you want I can prepare a testapp and give you access to it if that helps? Or I can show you in a google hangout?

**Reply by melohagan** [reply]
Hi @remcohh 

A test app would be fantastic if you could provide that.

**Reply by pankajjangid05** [reply]
Hi @melohagan 

I am also facing the same issue. can you please help me out?

For your context I am stating this issue here:

I created 2 tables one is Employee and the second one is skills both tables have many-to-many relationships. When I create a new employee it works fine but when I edit employees and update it, skills data is not updated but the rest of the employee data is updated in the table.

These are the table names: 
employee
skills
jt_employee_skills_Skills_employee

**Reply by melohagan** [reply]
Hey @pankajjangid05 

When you say 'edit' employee, do you mean directly in the data section, or in an edit detail screen? Some screenshots may help.

**Reply by pankajjangid05** [reply]
Hi @melohagan 

I mean not directly in the data section. I am editing in screen and I am sharing it with you screenshots below.


![Screenshot](https://user-images.githubusercontent.com/103931276/204535695-87cb7da0-966f-4e9f-8371-ede6e14d1ef6.png)


So basically I want to remove the **Docker** skill from the skills drop-down. I can add new skills but I am unable to remove them.  


![screencapture-localhost-10000-app-dev-ddc9cea9cb054d2781e6c9d099502a8e-2022-11-29-18_24_56](https://user-images.githubusercontent.com/103931276/204535289-3c9c0711-f530-44f2-808a-bcf4ac049fcc.png)

After : remove docker and save screenshot
![Screenshot (2)](https://user-images.githubusercontent.com/103931276/204537843-de84c6d0-7a2b-4fa8-afbf-de79ad817c7a.png)

**Reply by melohagan** [reply]
Hey @pankajjangid05 I came across a different issue with the same setup, which I've raised here: https://github.com/Budibase/budibase/issues/8882

**Comment by dominikjannis** [comment]
i am encountering the same issue, has this issue here been resolved?
